libs/utils/HashableDict.py

```diff
@@ -0,0 +1,3 @@
+class HashableDict(dict):
+    def __hash__(self):
+        return hash(tuple(sorted(self.items())))
```

Comment:
Isn't that already a class in some base library? Or what is special about it?

Reply:
Yes, good points, replaced with `frozendict` in e046b2a (the lib is on conda).